lxd/documentation.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"embed"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+)
+
+var documentationCmd = APIEndpoint{
+	Path: "documentation-config-options",
```

Comment:
Given that I dont (yet) see a need for a separate listener for the LXD docs endpoint itself (https://discourse.ubuntu.com/t/expose-the-full-lxd-documentation-website-as-part-of-a-lxd-deployment/37516), and even if we do, we should probably still expose this documentation yaml file over that listener anyway.

So my thinking here is that this `documentation-config-options` endpoint should probably be hierarchically under `/documentation`, but in a way it wont conflict with the full documentation URLs.

Reply:
cc @ru-fu

Reply:
Yes I agree, it would makes sense to do it this way

Reply:
As of now, in [this PR](https://github.com/canonical/lxd/pull/12098), because we have a custom listener for it, we must first a listener address/port (using `core.documentation_address`) and then the doc is accessible at this URL : https:<core.documentation_address>/ . The `/` at the end is not mandatory as there is a redirection rule

Reply:
> I'm not sure I agree that this YAML file provides config options documentation.

What would you say it provides? My aim is to avoid adding this long non-hierarchical API endpoint `documentation-config-options`, so open to suggestions.

Reply:
As I said, it provides the available config options. 
I think that for the UI, the list of all options and their grouping will be more important than the documentation for them.

Reply:
What is your proposal for api endpoint URL?

Reply:
Sounds good to me!

Reply:
Thanks. @gabrielmougard please can you go ahead and make that change as a priority so we can get this merged for LXD 5.17. Thanks

Reply:
sure